internal/providers/terraform/azure/storage_table.go

```diff
@@ -25,19 +24,16 @@ func newStorageTable(d *schema.ResourceData) schema.CoreResource {
 	region := d.Region
 
 	accountReplicationType := "LRS"
-	accountKind := "StorageV2"
+	hasCustomerManagedKey := false
 
 	if len(d.References("storage_account_name")) > 0 {
 		storageAccount := d.References("storage_account_name")[0]
+		accountReplicationType = storageAccount.Get("account_replication_type").String()
+		hasCustomerManagedKey = storageAccount.Get("customer_managed_key").Exists()
 
-		accountTier := storageAccount.Get("account_tier").String()
-		if strings.EqualFold(accountTier, "premium") {
-			logging.Logger.Warn().Msgf("Skipping resource %s. Storage Tables don't support %s tier", d.Address, accountTier)
-			return nil
+		if storageAccount.References("azurerm_storage_account_customer_managed_key.storage_account_id") != nil {
```

Comment:
This looks up to see if there's a separate [customer managed key resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account_customer_managed_key).